pkg/api/openshiftcluster.go

```diff
@@ -145,6 +145,8 @@ type OpenShiftClusterProperties struct {
 	UserAdminKubeconfig SecureBytes `json:"userAdminKubeconfig,omitempty"`
 
 	RegistryProfiles []*RegistryProfile `json:"registryProfiles,omitempty"`
+
+	SecretVersion string `json:"secretVersion,omitempty"`
```

Comment:
If it lives at a document level the only way to view the contents of `SecretVersion` is to query the cosmosDB directly.  The converters deal with an OCP cluster not at the document level but at the `pkg/api/openshiftcluster.go` level.  

The goal of having a `SecretVersion` is to ensure all encrypted fields within a document have been updated.  We wouldn't be able to query this in the admin API if it lives in `pkg/api/openshiftclusterdocument.go`.  

I can move it to a higher level in `pkg/api/openshiftcluster.go` instead of within `openshiftclusterproperties`.

Reply:
Why do we need this in the admin API?

Reply:
We can confirm all the documents are on a new secret version and remove old secrets.  Otherwise there's no way of knowing which document is encrypted with which version of the secret.

Reply:
Hm. I really think that properties is not the right place. Higher layer is better, but still this doesn't feel relevant to the cluster.

If we put it on the doc level - we can create a geneva action to report on secret version. But this sounds like a bit of overkill...

Reply:
Splitting the secret version out of this PR.